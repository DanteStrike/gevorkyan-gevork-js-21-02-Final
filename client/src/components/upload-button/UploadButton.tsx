import React from 'react';
import {LoadingOutlined, PlusOutlined} from '@ant-design/icons';

interface IUploadButtonProps {
  loading: boolean;
}
const style = {marginTop: 8};

function UploadButton({loading}: IUploadButtonProps) {
  return (
    <div>
      {loading ? <LoadingOutlined /> : <PlusOutlined />}
      <div style={style}>Upload</div>
    </div>
  );
}

export default React.memo(UploadButton);
