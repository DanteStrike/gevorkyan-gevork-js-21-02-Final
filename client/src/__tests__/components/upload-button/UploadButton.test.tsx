import React from 'react';
import {shallow} from 'enzyme';
import UploadButton from '../../../components/upload-button/UploadButton';

describe(`Component UploadButton should render correctly`, () => {
  it(`snapshot default test`, () => {
    const wrapper = shallow(<UploadButton loading={false} />);

    expect(wrapper).toMatchSnapshot();
  });

  it(`snapshot loading test`, () => {
    const wrapper = shallow(<UploadButton loading={false} />);

    expect(wrapper).toMatchSnapshot();
  });
});
