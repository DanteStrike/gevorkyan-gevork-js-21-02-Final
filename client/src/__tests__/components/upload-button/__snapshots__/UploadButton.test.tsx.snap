// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component UploadButton should render correctly snapshot default test 1`] = `
<div>
  <ForwardRef(PlusOutlined) />
  <div
    style={
      Object {
        "marginTop": 8,
      }
    }
  >
    Upload
  </div>
</div>
`;

exports[`Component UploadButton should render correctly snapshot loading test 1`] = `
<div>
  <ForwardRef(PlusOutlined) />
  <div
    style={
      Object {
        "marginTop": 8,
      }
    }
  >
    Upload
  </div>
</div>
`;
